evalml/pipelines/component_graph.py

```diff
@@ -288,7 +285,9 @@ def transform(self, X, y=None):
                 "Cannot call transform() on a component graph because the final component is not a Transformer."
             )
 
-        outputs = self._compute_features(self.compute_order, X, y, False)
+        outputs = self._compute_features(
+            self.compute_order, X, y, fit=False, evaluate_training=True
```

Comment:
I may be misunderstanding something here - why should `evaluate_training` be true during `transform`, but not during `fit`?

Reply:
@eccabay Good question! I think the language here is a bit confusing, but `evaluate_training` is a flag for determining whether or not we want to compute "training-only" components, such as the samplers or the DropRowsComponent when we are *not* fitting. 

We decided that there should be a difference between `transform`, which takes the pipeline and calls `transform` on every component vs `predict`, where we want predictions on unseen data and should not evaluate training-only components. This enables us to keep our current behavior, but also allow users to use pipelines as just a sequence of transformations.

Does that make sense? (or did I confuse you further?) 😄

Reply:
Ah yeah, that makes a lot of sense. Thanks! 😃